pennylane/measurements/mutual_info.py

```diff
@@ -76,4 +79,17 @@ def circuit_mutual(x):
 
     wires0 = qml.wires.Wires(wires0)
     wires1 = qml.wires.Wires(wires1)
-    return MeasurementProcess(MutualInfo, wires=[wires0, wires1], log_base=log_base)
+    return _MutualInfo(MutualInfo, wires=[wires0, wires1], log_base=log_base)
+
+
+class _MutualInfo(StateMeasurement):
+    """Measurement process that returns the mutual information."""
+
+    def process_state(self, state: Sequence[complex], wires: Wires):
+        return qml.math.mutual_info(
+            state,
+            indices0=[self.wires[0]],
+            indices1=[self.wires[1]],
+            c_dtype="complex128",
```

Comment:
In light of the recent JAX issue where the result is unexpectedly cast to double precision, we want to avoid hard-coding in a dtype. Can we use the dtype of the state here?

Reply:
Sure! 👍